include/SFML/Graphics/Sprite.hpp

```diff
@@ -75,7 +64,7 @@ class SFML_GRAPHICS_API Sprite : public Drawable, public Transformable
     /// \see setTexture, setTextureRect
     ///
     ////////////////////////////////////////////////////////////
-    Sprite(const Texture& texture, const IntRect& rectangle);
+    Sprite(const Texture& texture, const IntRect& rectangle = IntRect());
```

Comment:
I don't see a reason to remove the other constructor and change this (and especially not adding an implicit conversion). The existing behavior is explicit enough; both constructors have different semantics and are documented as such.

Maybe focus on `sf::Rect` changes in this PR, and if you want to discuss other things, consider opening a new issue or PR 🙂

Reply:
Agreed. Done